packages/pigeon/lib/pigeon_lib.dart

```diff
@@ -628,15 +648,23 @@ class _RootBuilder extends dart_ast_visitor.RecursiveAstVisitor<Object?> {
           .firstWhere((e) => e is dart_ast.TypeName) as dart_ast.TypeName;
       argType = typeName.name.name;
       isNullable = typeName.question != null;
+      argTypeArguments = typeAnnotationsToTypeArguments(typeName.typeArguments);
     }
     final bool isAsynchronous = _hasMetadata(node.metadata, 'async');
     if (_currentApi != null) {
       _currentApi!.methods.add(Method(
           name: node.name.name,
-          returnType: node.returnType.toString(),
-          argType: argType,
-          isReturnNullable: node.returnType!.question != null,
-          isArgNullable: isNullable,
+          returnType: Field(
+              name: '',
+              dataType: node.returnType.toString(),
+              typeArguments: typeAnnotationsToTypeArguments(
+                  (node.returnType as dart_ast.NamedType?)!.typeArguments),
+              isNullable: node.returnType!.question != null),
```

Comment:
And then these would be a single parameter to Field.

Reply:
Yep, just mentioned returnType.  Do you want me to file an issue to revisit this?  Otherwise I'll can keep an eye on this.  Check out `TypedEntity`.

Reply:
I called this out as one of the motivations for adding `TypeDeclaration`, so I'm not sure why it isn't using it now.